lib/lago/__init__.py

```diff
@@ -464,46 +465,67 @@ def _create_disk(
 
             disk_filename = '%s_%s.%s' % (name, spec['name'], spec['format'])
             disk_path = self.paths.images(disk_filename)
+
             if spec['type'] == 'template':
-                if template_store is None or template_repo is None:
-                    raise RuntimeError('No templates directory provided')
+                template_type = spec.get('template_type', 'repository')
+                if template_type == 'repository':
+                    if template_store is None or template_repo is None:
+                        raise RuntimeError('No templates directory provided')
+
+                    template = template_repo.get_by_name(spec['template_name'])
+                    template_version = template.get_version(
+                        spec.get('template_version', None)
+                    )
 
-                template = template_repo.get_by_name(spec['template_name'])
-                template_version = template.get_version(
-                    spec.get('template_version', None)
-                )
+                    if template_version not in template_store:
+                        LOGGER.info(
+                            log_utils.log_always(
+                                "Template %s not in cache, downloading"
+                            ) % template_version.name,
+                        )
+                        template_store.download(template_version)
+                    template_store.mark_used(
+                        template_version, self.paths.uuid())
 
-                if template_version not in template_store:
-                    LOGGER.info(
-                        log_utils.log_always(
-                            "Template %s not in cache, downloading"
-                        ) % template_version.name,
+                    disk_metadata.update(
+                        template_store.get_stored_metadata(
+                            template_version,
+                        ),
                     )
-                    template_store.download(template_version)
-                template_store.mark_used(template_version, self.paths.uuid())
-
-                disk_metadata.update(
-                    template_store.get_stored_metadata(
-                        template_version,
-                    ),
-                )
 
-                base = template_store.get_path(template_version)
-                qemu_img_cmd = [
-                    'qemu-img', 'create', '-f', 'qcow2', '-b', base, disk_path
-                ]
+                    base = template_store.get_path(template_version)
+                    qemu_img_cmd = [
+                        'qemu-img', 'create',
+                        '-f', 'qcow2',
+                        '-b', base, disk_path
+                    ]
 
+                elif template_type == 'file':
```

Comment:
Probably qcow is better here too, or snapshot, file is too broad and can be easily confused with the file image provider